swift-extensions/UIImageViewExtensions.swift

```diff
@@ -1,113 +1,197 @@
 
-        URLSession.shared.dataTask(with: URL) {[weak self] data, response, error in
-            if let httpURLResponse = response as? HTTPURLResponse, httpURLResponse.statusCode == 200,
-                let mimeType = response?.mimeType, mimeType.hasPrefix("image"),
-                let data = data, error == nil,
-                let image = UIImage(data: data) {
+        if isImageCacheEnabled, let cachedImage = imageCache.object(forKey: url.absoluteString as NSString) {
+            imageView.restoreContentMode()
+            imageView.image = cachedImage
+        } else {
+            MrFreeze().freeze(objectToFreeze: cancellableManager)
 
+            let dataTask = URLSession.shared.dataTask(with: url) { [weak imageView, weak self] data, response, error in
                 DispatchQueue.main.async {
-                    guard let self = self else { return }
-                    self.contentMode = objc_getAssociatedObject(self, USER_CONTENT_MODE_KEY) as? ContentMode ?? .scaleToFill
-                    self.image = image
-
-                    if let onSuccess = imageFlow.onSuccess {
-                        self.observeImageFlow(cancellableManager: cancellableManager, imageFlowPublisher: onSuccess)
+                    guard let imageView = imageView else { return }
+                    if let httpURLResponse = response as? HTTPURLResponse, httpURLResponse.statusCode == 200,
+                        let mimeType = response?.mimeType, mimeType.hasPrefix("image"),
+                        let data = data, error == nil,
+                        let image = UIImage(data: data) {
+
+                        self?.imageCache.setObject(image, forKey: url.absoluteString as NSString, cost: image.cacheCost)
+                        imageView.restoreContentMode()
+                        imageView.image = image
                     }
                 }
-            } else if let onError = imageFlow.onError {
-                DispatchQueue.main.async {
-                    self?.observeImageFlow(cancellableManager: cancellableManager, imageFlowPublisher: onError)
-                }
             }
-        }.resume()
-    }
-
-    public var imageResourceContentMode: ContentMode {
-        set(value) {
-            objc_setAssociatedObject(self, USER_IMAGERESOURCE_CONTENT_MODE_KEY, value, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN)
-        }
-        get {
-            return objc_getAssociatedObject(self, USER_IMAGERESOURCE_CONTENT_MODE_KEY) as? ContentMode ?? .center
+            cancellableManager.add(cancellable: dataTask)
+            dataTask.resume()
         }
     }
+}
 
-    public var placeholderContentMode: ContentMode {
-        set(value) {
-            objc_setAssociatedObject(self, USER_PLACEHOLDER_CONTENT_MODE_KEY, value, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN)
-        }
+extension URLSessionTask: Cancellable {}
+
+extension UIImage {
+    var cacheCost: Int {
```

Comment:
Ça ne devrait pas être private comme extension ça?

Reply:
:true: sauf que c'est pas public par défaut, juste internal.

Reply:
As it is not public, its considered as internal to the pod